### ISSUE TITLE
Some comparison with Scala?

### ISSUE THREAD

**Issue by ppKrauss** [parsifal-47/sisal-is#11]
I was a 1990's user of SISAL for dataflow-machine simulators... But I can't remember. In nowadays I am using Scala, that have similar features and is so popular. There are some document with some comparisons?

----

Example of similar constructions to be compared: FFT 2D, [Scala](https://rosettacode.org/wiki/Fast_Fourier_transform#Scala) and [SISAL](https://github.com/parsifal-47/sisal-is/blob/master/examples/complex/dft2.sis).  <br/>PS: looking for simple cases; a [complete and optimized Scala implementation](https://scala-lms.github.io/tutorials/fft.html) is not simple.

**Comment by parsifal-47** [comment]
Good question, I haven't seen such document, but I guess we can make a comparison, should be interesting to read, thanks!

**Comment by ppKrauss** [comment]
Hi @parsifal-47 , thanks your fast reply (!). Ok I will check some examples that we can compare with rosettacode.org.

Another interest about SISAL is the result of compiler, the old "intermediary language" that is like a table representing the dataflow.  Today an important application is to generate or optimize complex [DAGs](https://en.wikipedia.org/wiki/Directed_acyclic_graph) for  [coarse grain](https://en.wikipedia.org/wiki/Granularity_(parallel_computing)) parallellism (process orchestration), for tools like [Apache AirFlow](https://airflow.apache.org/).

**Comment by parsifal-47** [comment]
yes, the problem with Sisal IR is its inflexibility, if you look how loops defined, they are too specific to Sisal. Another example: closures and namespacing missing in original Sisal. But some parts are really interesting, for example loop expressions and reductions.

**Comment by ppKrauss** [comment]
> yes, the problem with Sisal IR is its inflexibility (...)

This "Sisal IR" is the equivalent of the old compiler output IF1? <br/> IF1 = "dataflow graph in Intermediary Form 1". 

The IF1 is "easy" to translate (more one parse step) to a generic dataflow... Why you say that is *"too specific"*?

----

About examples comparing with Scala, we can post at Rosetta (!), see for example the Factorial example: https://rosettacode.org/wiki/Factorial#Sisal
and it will be compared with any other language.   Do you have examples that use closures and namespacing?
